src/Data/List/Zipper/Properties.agda

```diff
@@ -0,0 +1,131 @@
+     ≡⟨ sym (reverse-++-commute (List.reverse ctx) val) ⟩
+   List.reverse (List.reverse ctx List.++ val)
+     ∎
+
+
+-- Properties of the insertion functions
+------------------------------------------------------------------------
+
+ -- _ˡ++_ properties
+
+ toList-ˡ++-commute : ∀ xs (zp : Zipper A) → toList (xs ˡ++ zp) ≡ xs List.++ toList zp
+ toList-ˡ++-commute xs (mkZipper ctx val) = begin
+   List.reverse (ctx List.++ List.reverse xs) List.++ val
+     ≡⟨ cong (List._++ _) (reverse-++-commute ctx (List.reverse xs)) ⟩
+   (List.reverse (List.reverse xs) List.++ List.reverse ctx) List.++ val
+     ≡⟨ ++-assoc (List.reverse (List.reverse xs)) (List.reverse ctx) val ⟩
+   List.reverse (List.reverse xs) List.++ List.reverse ctx List.++ val
+     ≡⟨ cong (List._++ _) (reverse-involutive xs) ⟩
+   xs List.++ List.reverse ctx List.++ val
+     ∎
+
+ ˡ++-assoc : ∀ xs ys (zp : Zipper A) → xs ˡ++ (ys ˡ++ zp) ≡ (xs List.++ ys) ˡ++ zp
+ ˡ++-assoc xs ys (mkZipper ctx val) = cong (λ ctx → mkZipper ctx val) $ begin
+   (ctx List.++ List.reverse ys) List.++ List.reverse xs
+     ≡⟨ ++-assoc ctx _ _ ⟩
+   ctx List.++ List.reverse ys List.++ List.reverse xs
+     ≡⟨ cong (ctx List.++_) (sym (reverse-++-commute xs ys)) ⟩
+   ctx List.++ List.reverse (xs List.++ ys)
+     ∎
+
+ -- _ʳ++_ properties
+
+ ʳ++-assoc : ∀ xs ys (zp : Zipper A) → xs ʳ++ (ys ʳ++ zp) ≡ (ys List.++ xs) ʳ++ zp
+ ʳ++-assoc xs ys (mkZipper ctx val) =  cong (λ ctx → mkZipper ctx val) $ begin
+   List.reverse xs List.++ List.reverse ys List.++ ctx
+     ≡⟨ sym (++-assoc (List.reverse xs) (List.reverse ys) ctx) ⟩
+   (List.reverse xs List.++ List.reverse ys) List.++ ctx
+     ≡⟨ cong (List._++ ctx) (sym (reverse-++-commute ys xs)) ⟩
+   List.reverse (ys List.++ xs) List.++ ctx
+     ∎
+
+ -- _++ˡ_ properties
+
+ ++ˡ-assoc : ∀ xs ys (zp : Zipper A) → zp ++ˡ xs ++ˡ ys ≡ zp ++ˡ (ys List.++ xs)
+ ++ˡ-assoc xs ys (mkZipper ctx val) =  cong (mkZipper ctx) $ sym $ ++-assoc ys xs val
+
+ -- _++ʳ_ properties
+
+ toList-++ʳ-commute : ∀ (zp : Zipper A) xs → toList (zp ++ʳ xs) ≡ toList zp List.++ xs
+ toList-++ʳ-commute (mkZipper ctx val) xs = begin
+   List.reverse ctx List.++ val List.++ xs
+     ≡⟨ sym (++-assoc (List.reverse ctx) val xs) ⟩
+   (List.reverse ctx List.++ val) List.++ xs
+     ∎
+
+ ++ʳ-assoc : ∀ xs ys (zp : Zipper A) → zp ++ʳ xs ++ʳ ys ≡ zp ++ʳ (xs List.++ ys)
+ ++ʳ-assoc xs ys (mkZipper ctx val) = cong (mkZipper ctx) $ ++-assoc val xs ys
+
+
+-- List-like operations indeed correspond to their counterparts
+------------------------------------------------------------------------
+
+module _ {a b} {A : Set a} {B : Set b} where
+
+ toList-map-commute : ∀ (f : A → B) zp → toList (map f zp) ≡ List.map f (toList zp)
+ toList-map-commute f zp@(mkZipper ctx val) = begin
+   List.reverse (List.map f ctx) List.++ List.map f val
+     ≡⟨ cong (List._++ _) (sym (reverse-map-commute f ctx)) ⟩
+   List.map f (List.reverse ctx) List.++ List.map f val
+     ≡⟨ sym (map-++-commute f _ val) ⟩
```

Comment:
Just to note that this `_` wasn't inferable by Agda 2.5.3. I've fixed it in 37baa09.

Hopefully Travis will catch this next time now that it's back up and running.

Reply:
It did catch it this time because I rebased the branch against master after fixing the builds.
However the fact the PR had been marked as failing since it was created made it quite hard
to notice the failure! :D